### PR TITLE
Extract Renderer layer

### LINKED ISSUE #92: Extract Renderer code to a single class/interface
Currently there's a `Box.Draw` method that's implemented in a relative small number of classes.

For further extensibility, I'd like these methods to be extracted to the `IRenderer` interface and `WpfRenderer` class using the Visitor pattern (or something alike).

In future, we'll likely extract `Box.RenderGeometry` (that has the same form as the `Draw` method) to another `IRenderer` implementation, `GeometryRenderer`. That will also open a way for alternate implementations, such as `DirectXRenderer` and `AvaloniaRenderer`.

### PR DESCRIPTION
**Pull request by ForNeVeR**
Closes #92.

**Things remaining to do**:

- [x] extract `RenderGeometry` method to `GeometryRenderer` class
- [x] add some tests! (with maybe a `TestRenderer`?)

**Open questions**:

- [x] I'd like to uncouple `IElementRenderer` from WPF's `Transform` type. Maybe replace it with our own abstraction that will be converted into WPF `Transform` or just `Translate` / `Rotate` method calls on the `GeometryRender`?
- [x] ~~I don't like the first argument of `RenderGlyphRun` (because it's a lambda). Maybe we could somehow replace that with `GlyphRun` and scale it in the renderer code?~~
  _Added a TODO comment for that, too hard to implement at the moment._

**Impact**:

As I said in #92, this opens a supported way to implement alternate renderders for DirectX, Avalonia and whatnot. There's still more work to do with `GlyphRun` replacement, but I think we'll do that later.

### CODE REVIEW

File: src/WpfMath/Rendering/GeometryRenderer.cs [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+namespace WpfMath.Rendering
+{
+    /// <summary>A renderer that renders the elements to a provided <see cref="GeometryGroup"/> instance.</summary>
+    public class GeometryRenderer : IElementRenderer
+    {
+        private readonly double _scale;
+        private readonly GeometryGroup _geometry;
+
+        public GeometryRenderer(double scale, GeometryGroup geometry)
```
Comment (ForNeVeR) [inline src/WpfMath/Rendering/GeometryRenderer.cs]: - [ ] pass `scale` as a second element everywhere

File: src/WpfMath/Rendering/GeometryRenderer.cs [hunk 2]
```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Windows;
+using System.Windows.Media;
+
+namespace WpfMath.Rendering
+{
+    /// <summary>A renderer that renders the elements to a provided <see cref="GeometryGroup"/> instance.</summary>
+    public class GeometryRenderer : IElementRenderer
```
Comment (ForNeVeR) [inline src/WpfMath/Rendering/GeometryRenderer.cs]: - [ ] → `GeometryElementRenderer`?

File: src/WpfMath/CharBox.cs [hunk 3]
```diff
@@ -22,31 +23,20 @@ public CharInfo Character
             private set;
         }
 
-        private GlyphRun GetGlyphRun(double scale, double x, double y)
+        internal GlyphRun GetGlyphRun(double scale, double x, double y)
```
Comment (gsomix) [inline src/WpfMath/CharBox.cs]: `GlyphRun` is WPF class. Please don't forget create issue to move this part to WPF renderer.

Comment (ForNeVeR) [reply src/WpfMath/CharBox.cs]: - [ ] also, this particular line (changing of `private` to `internal`) seem unnecessary?

Thanks for your review!

Comment (ForNeVeR) [reply src/WpfMath/CharBox.cs]: > this particular line (changing of `private` to `internal`) seems to be unnecessary?

Nope, it doesn't. This method is called from the test code.

**Comment by alexreg** [comment]
Good job, guys!

**Pull request merged**